bn_mp_mul.c

```diff
@@ -3,6 +3,13 @@
 /* LibTomMath, multiple-precision integer library -- Tom St Denis */
 /* SPDX-License-Identifier: Unlicense */
 
+#define _MP_ENABLED_TEST        ,
+#define _MP_ENABLED_TEST1       ,
+#define MP_ENABLED(x)           _MP_ENABLED1(x)
+#define _MP_ENABLED1(x)         _MP_ENABLED2(_MP_ENABLED_TEST##x)
+#define _MP_ENABLED2(x)         _MP_ENABLED3(x 1, 0)
+#define _MP_ENABLED3(x, y, ...) y
```

Comment:
@czurnieden @sjaeckel How do you like something like this.

The macro will evaluate to 1 if the macro is either defined as empty or defined as 1.

Reply:
The original macros were meant to exclude code from compiling, from going into the binary in the first place. This patch keeps the branches in and hopes for the compiler to optimize them away.
Is it the case? Have you checked the dumps of GCC and clang? What is left if compiled with `-Os`?

Reply:
This will always work. Every compiler optimizes `if (0)` away. For example the Linux kernel does that too. It is actually a quite common practice. The advantage of this is that you always get the typechecking.

Reply:
The only situation where you have to go back to #ifdefs is if the definition is missing or changes depending on macros.